docker_copy.go

```diff
@@ -0,0 +1,172 @@
+// Copy of https://github.com/docker/docker/blob/master/pkg/stdcopy/stdcopy.go
```

Comment:
Afaik we need to add their license as well. Might make sense to move the whole thing to a subpackage to make it clearer.